src/wp-admin/update.php

```diff
@@ -163,7 +163,12 @@
 		$type  = 'upload'; // Install plugin type, From Web or an Upload.
 
 		$upgrader = new Plugin_Upgrader( new Plugin_Installer_Skin( compact( 'type', 'title', 'nonce', 'url' ) ) );
-		$result   = $upgrader->install( $file_upload->package );
+
+		$args = array(
+			'overwrite_package' => ! empty( $_GET['overwrite'] ),
```

Comment:
May be good to make the query arg a bit longer/more specific? Perhaps something like `$_GET['overwrite'] === 'ovewrite-uploaded-plugin'`.